### ISSUE TITLE
QA Process - Discovery - Automated visual regression testing

### ISSUE THREAD

**Issue by micahchiang** [department-of-veterans-affairs/vets-design-system-documentation#2218]
## Description
In order to increase the level of confidence we have in our components, we should include automated visual regression testing in our QA process. 

## Considerations
- Storybook has some guidance on [using Chromatic for regression testing](https://storybook.js.org/docs/react/writing-tests/visual-testing). It's in early access right now but may help point us in the right direction
- There is an experimental Chromatic [Visual testing addon for Storybook](https://www.chromatic.com/docs/visual-testing-addon/) that we can try
- Maybe we can use cypress to do this since it already runs in CI

## Tasks
- [ ] Discovery work on options for visual regression testing

## Acceptance Criteria
- [ ] A summary of potential options and a recommendation are added to this ticket as a comment

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2218?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc

**Comment by Andrew565** [comment]
I've completed my discovery efforts, [overview available here](https://docs.google.com/document/d/1iEO-M-d24s3O4GVHksZxl3wxx0VI9Kx0zaicTpIn_JI/edit?usp=sharing).

TL;DR: If we want a local option with no UI, Loki works well. If we don't mind it being cloud-based, Chromatic is truly the better option all around.

**Comment by Andrew565** [comment]
Longer version for anyone who is interested:

I tested six total options. The first four I was unable to get to work, half because they’re too old, half because we are still using the 6.X version of Storybook instead of the 7.X version. 

FWIW, I did try a ‘quick’ upgrade of Storybook to version 7.6, and it did not go well. The automigration scripts did a lot, but there were still some weird errors that came up that I wasn’t able to quickly find solutions for, so I think the upgrade process will require dedicated time.

Of the remaining two testing options that I could get to work, Loki was easy to setup and runs locally, but requires Docker (set itself up without problems) and only generates the difference images without a UI for reviewing them.

Chromatic requires an internet connection to use, but was equally easy to setup (used my personal Github for the test), the UI was easy to use and included a tutorial for the first walkthrough which made it easy to start, and we already know it will integrate with our CI.

All of the options tested made use of Storybook in some way, no options were tested that made use of Cypress instead, though that remains a possible option.
